src/operations/liquidity_pool_deposit.js

```diff
@@ -0,0 +1,59 @@
+import isUndefined from 'lodash/isUndefined';
+import xdr from '../generated/stellar-xdr_generated';
+
+/**
+ * Creates a liquidity pool deposit operation.
+ *
+ * @function
+ * @alias Operation.liquidityPoolDeposit
+ * @see https://developers.stellar.org/docs/start/list-of-operations/#liquidity-pool-deposit
+ *
+ * @param {object} opts - Options object
+ * @param {string} opts.liquidityPoolId - The liquidity pool ID.
+ * @param {string} opts.maxAmountA - Maximum amount of first asset to deposit.
+ * @param {string} opts.maxAmountB - Maximum amount of second asset to deposit.
+ * @param {number|string|BigNumber|Object} opts.minPrice -  Minimum depositA/depositB price.
+ * @param {number} opts.minPrice.n - If `opts.minPrice` is an object: the price numerator
+ * @param {number} opts.minPrice.d - If `opts.minPrice` is an object: the price denominator
+ * @param {number|string|BigNumber|Object} opts.maxPrice -  Maximum depositA/depositB price.
+ * @param {number} opts.maxPrice.n - If `opts.maxPrice` is an object: the price numerator
+ * @param {number} opts.maxPrice.d - If `opts.maxPrice` is an object: the price denominator
+ * @param {string} [opts.source] - The source account for the operation. Defaults to the transaction's source account.
+ *
+ * @returns {xdr.Operation} The resulting operation (xdr.LiquidityPoolDepositOp).
+ */
+export function liquidityPoolDeposit(opts = {}) {
+  const attributes = {};
+  if (!opts.liquidityPoolId) {
+    throw new TypeError('liquidityPoolId argument is required');
+  }
+  attributes.liquidityPoolId = xdr.PoolId.fromXDR(opts.liquidityPoolId, 'hex');
+
+  if (isUndefined(opts.maxAmountA)) {
```

Comment:
I think using `Operation.isValidAmount` would be a better validation here.

Reply:
TIL! Changed to `Operation.isValidAmount` in https://github.com/stellar/js-stellar-base/pull/454/commits/43de4f1ed5d0abd37faa3b28401d8f5b632e52a3.